src/lib/Syntax.hs

```diff
@@ -146,119 +121,123 @@ scalarTableBaseType t = case t of
   BaseTy b  -> b
   _         -> error $ "Not a scalar table: " ++ show t
 
-noEffect :: Effect
-noEffect = Effect mempty Nothing
-
-isPure :: Effect -> Bool
-isPure (Effect eff Nothing) | eff == mempty = True
-isPure _ = False
-
-type LTVar = VarP (LorT Type Kind)
-type ModuleType = ([LTVar], [LTVar])
-data ModuleP body = Module (Maybe BlockId) ModuleType body  deriving (Show, Eq)
-
 -- === front-end language AST ===
 
-data FExpr = FDecl FDecl FExpr
-           | FVar Var
-           | FPrimExpr (PrimExpr Type FExpr FLamExpr)
-           | Annot FExpr Type
-           | SrcAnnot FExpr SrcPos -- TODO: make mandatory?
-             deriving (Eq, Show, Generic)
-
-type Pat = RecTree Var
-data FLamExpr = FLamExpr Pat FExpr  deriving (Show, Eq, Generic)
-type SrcPos = (Int, Int)
-
--- TODO: Unify LetMono and LetPoly. The only real difference is having a type
---       annotation attached and we can handle this with bidirectional try inference.
-data FDecl = LetMono Pat FExpr
-           | LetPoly Var FTLam
-           | TyDef TVar Type
-             deriving (Show, Eq, Generic)
-
-type Var  = VarP Type
-data FTLam = FTLam [TVar] [TyQual] FExpr  deriving (Show, Eq, Generic)
-type FModule = ModuleP [FDecl]
-
-data RuleAnn = LinearizationDef Name    deriving (Show, Eq, Generic)
+type UExpr = WithSrc UExpr'
```

Comment:
Might be nice to preserve the `F*` naming convention instead of `U` (which I suppose comes from unified and made sense when there were two frontend syntaxes, but now there is only one).

Reply:
Actually the `U` is for "untyped"! We used to only have `UExpr`, then `FExpr` came in alongside it. Now we're back to `UExpr`. I don't have strong feelings either way.